private/buf/cmd/protoc-gen-buf-breaking/breaking.go

```diff
@@ -43,7 +44,10 @@ const (
 
 // Main is the main.
 func Main() {
-	protoplugin.Main(protoplugin.HandlerFunc(handle))
+	protoplugin.Main(
+		protoplugin.HandlerFunc(handle),
+		protoplugin.WithExtensionTypeResolver(protoencoding.EmptyResolver),
```

Comment:
I think we want some kind of comment here (and for lint) that explains why we are passing in `EmptyResolver` here, e.g.

```
// Passing in an `EmptyResolver` to protoplugin for unmarshalling instead of
// defaulting to the protoregistry.GlobalTypes. This prevents marking unused extensions
// that are resolved through protoregistry.GlobalTypes as used.
protoplugin.WithExtensionTypeResolver(protoencoding.EmptyResolver)
```

(Feel free the massage the wording of the comment ^^")

Reply:
Done, I reworded it a little bit and linked the issue as well.